### PR TITLE
Fix#1347 Fix VaInfinite scroll

### PR DESCRIPTION
**Pull request by RVitaly1978**
## Description
close: #1347

Changes:
- [x] - rewrite component on Composition API
- [x] - add events: `onload` and `onerror`
- [x] - update examples

chrome-capture

chrome-capture (1)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

File: packages/ui/src/components/va-infinite-scroll/hooks/useScroll.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    if (props.reverse) {
+      element.value.scrollTop = element.value.scrollHeight
+    }
+
+    addListener()
+  })
+
+  onBeforeUnmount(() => {
+    if (props.disabled) { return }
+
```
Comment (m0ksem) [inline packages/ui/src/components/va-infinite-scroll/hooks/useScroll.ts:43]:
```suggestion
```

I think there is no need for this check.

File: packages/ui/src/components/va-infinite-scroll/hooks/useScroll.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    removeListener()
+  })
+
+  return {
+    addListener,
+    removeListener,
```
Comment (m0ksem) [inline packages/ui/src/components/va-infinite-scroll/hooks/useScroll.ts:49]: Please, mark it as Scroll Listener.

```suggestion
    addScrollListener,
    removeScrollListener,
```

File: packages/docs/src/page-configs/ui-elements/infinite-scroll/examples/Default.vue [hunk 3]
```diff
@@ -6,6 +6,7 @@
     >
       <va-infinite-scroll
         :load="appendRecordsAsync"
+        :offset="40"
```
Comment (m0ksem) [inline packages/docs/src/page-configs/ui-elements/infinite-scroll/examples/Default.vue:9]: Why do we need to offset in default demo? I think the default demo should work without additional props.

As I user, I want components "just work".

```suggestion
```

File: packages/docs/src/page-configs/ui-elements/infinite-scroll/examples/Reverse.vue [hunk 4]
```diff
@@ -6,6 +6,7 @@
     >
       <va-infinite-scroll
         :load="appendRecordsAsync"
+        :offset="40"
```
Comment (m0ksem) [inline packages/docs/src/page-configs/ui-elements/infinite-scroll/examples/Reverse.vue:9]: Let's better have additional demo with offset.

```suggestion
```

File: packages/ui/src/components/va-infinite-scroll/VaInfiniteScroll.vue [hunk 5]
```diff
@@ -1,30 +1,26 @@
 <template>
   <component
-    :is="$props.tag"
+    :is="tag"
```
Comment (m0ksem) [inline packages/ui/src/components/va-infinite-scroll/VaInfiniteScroll.vue:3]:
```suggestion
    :is="$props.tag"
```

$props can help us separate props, computed and data properties in template. Do you see any reasons to remove this $props (Maybe some issues with types or smth like this)?

**Review by m0ksem — CHANGES_REQUESTED** [review]

File: packages/ui/src/components/va-infinite-scroll/hooks/useScroll.ts (hunk 2 above)
Comment (RVitaly1978) [reply packages/ui/src/components/va-infinite-scroll/hooks/useScroll.ts:49]: fixed it

File: packages/ui/src/components/va-infinite-scroll/hooks/useScroll.ts (hunk 1 above)
Comment (RVitaly1978) [reply packages/ui/src/components/va-infinite-scroll/hooks/useScroll.ts:43]: fixed it

File: packages/ui/src/components/va-infinite-scroll/VaInfiniteScroll.vue (hunk 5 above)
Comment (RVitaly1978) [reply packages/ui/src/components/va-infinite-scroll/VaInfiniteScroll.vue:3]: returned everything back ))

File: packages/docs/src/page-configs/ui-elements/infinite-scroll/examples/Default.vue (hunk 3 above)
Comment (RVitaly1978) [reply packages/docs/src/page-configs/ui-elements/infinite-scroll/examples/Default.vue:9]: returned everything back ))

File: packages/docs/src/page-configs/ui-elements/infinite-scroll/examples/Reverse.vue (hunk 4 above)
Comment (RVitaly1978) [reply packages/docs/src/page-configs/ui-elements/infinite-scroll/examples/Reverse.vue:9]: returned everything back ))

**Review by m0ksem — APPROVED** [review]

File: packages/ui/src/components/va-infinite-scroll/VaInfiniteScroll.vue (hunk 5 above)
Comment (Derranion) [reply packages/ui/src/components/va-infinite-scroll/VaInfiniteScroll.vue]: Just wanted to add that IDE highlights props \ data \ computed with a different color but yeah adding `props` may make it easier to review.

**Pull request merged**